pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -41,6 +41,10 @@ func ValidateClusterConfig(cfg *ClusterConfig) error {
 		return fmt.Errorf("iam.withOIDC must be enabled explicitly for iam.serviceAccounts to be created")
 	}
 
+	if IsDisabled(cfg.IAM.WithOIDC) && len(cfg.IAM.OIDCClientIDList) > 0 {
```

Comment:
We may want to move this validation to [`NewUtilsAssociateIAMOIDCProviderLoader`](https://github.com/weaveworks/eksctl/blob/f9e384a6efaa120a25624049c79053c560d413a5/pkg/ctl/cmdutils/configfile.go#L438). WDYT @cPu1, @martina-if?

@almariah, to provide some context, `eksctl` accepts input from both:

- CLI flags/args,
- `ClusterConfig` YAML file,

and then is supposed to merge this input, and then validate the end-result.

Reply:
@marccarre I figured it, in case of `validateWithConfigFile` we don't need this validation because

https://github.com/weaveworks/eksctl/blob/5817e7632b33af9852d87a1e8ff4be231613ff21/pkg/ctl/cmdutils/configfile.go#L465-L470